core/roboconf-agent/src/main/java/net/roboconf/agent/internal/Agent.java

```diff
@@ -425,8 +450,8 @@ public void setScopedInstancePath( String scopedInstancePath ) {
 	 * @param ipAddress the ipAddress to set
 	 */
 	public void setIpAddress( String ipAddress ) {
-		this.ipAddress = ipAddress;
-		this.logger.finer( "New IP address set in the agent: " + ipAddress );
+		if(! Utils.isEmptyOrWhitespaces(ipAddress)) this.ipAddress = ipAddress;
+		this.logger.finer( "New IP address set in the agent (if not empty): " + ipAddress );
```

Comment:
Put the logger invocation in the "if" block.
Otherwise, when reading the logs, we will not be able to determine whether the IP address was updating or not.